src/dparse/lexer.d

```diff
@@ -2198,61 +2193,85 @@ unittest
     assert (toks(`'\xXX'`).messages[0] == DLexer.Message(1,4,"Error: 2 hex digits expected.",true));
 }
 
-version (iasm64NotWindows)
+version (X86_64)
 {
+    private ulong pcmpestri(ubyte flags, chars...)(const ubyte* bytes) pure nothrow
+        @trusted @nogc if (chars.length <= 8)
+    {
+        enum constant = ByteCombine!chars;
+        enum charsLength = chars.length;
+
+        version (DigitalMars)
+        {
+            asm pure nothrow @nogc
+            {
+                naked;
+            }
+            version (Windows) // `bytes` in RCX
+                asm pure nothrow @nogc { movdqu XMM1, [RCX]; }
+            else // `bytes` in RDI
+                asm pure nothrow @nogc { movdqu XMM1, [RDI]; }
+            asm pure nothrow @nogc
+            {
+                mov R10, constant;
+                movq XMM2, R10;
+                mov RAX, charsLength;
+                mov RDX, 16;
+                pcmpestri XMM2, XMM1, flags;
+                mov RAX, RCX;
+                ret;
+            }
+        }
+        else // LDC/GDC
+        {
+            ulong result;
+            asm pure nothrow @nogc
+            {
+                `movdqu    %1, %%xmm1
+                 movq      %3, %%xmm2
+                 pcmpestri %5, %%xmm1, %%xmm2`
+                : "=c" (result)   // %0: pcmpestri result in RCX, to be stored into `result`
+                : "m" (*bytes),   // %1: address of `bytes` string
+                  "d" (16),       // %2: length of `bytes` head in XMM1, as pcmpestri input in EDX
+                  "r" (constant), // %3: max 8 `chars` to load into GP register, then XMM2
+                  "a" (charsLength), // %4: length in XMM2, as pcmpestri input in EAX
+                  "i" (flags)     // %5: `flags` immediate
+                : "xmm1", "xmm2"; // clobbered registers
+            }
+            return result;
+        }
+    }
+
     /**
      * Skips between 0 and 16 bytes that match (or do not match) one of the
      * given $(B chars).
      */
-    void skip(bool matching, chars...)(const ubyte*, ulong*, ulong*) pure nothrow
+    void skip(bool matching, chars...)(const ubyte* bytes, ulong* pindex, ulong* pcolumn) pure nothrow
```

Comment:
Can these attributes not be inferred?

Reply:
Most likely, but I didn't want to mix in other refactorings (the pointers would have been gone first and replaced by refs obviously...).

Reply:
I think avoiding refactoring solely on principle is a good way to kill progress.

Reply:
I don't avoid it by principle, I decide from time to time. The point here is clearly asm, not tampering with a codebase I've seen first yesterday and don't plan to contribute to regularly.